index.js

```diff
@@ -69,16 +75,26 @@ function writeDescription (schema, options) {
         : typeMatch || schema.type
       }}`
   }
-  return `  *${description}
-  * @${options.objectTagName || 'typedef'}${type}${schema.title
+  let descriptionLine = ''
+  if (description || options.emptyDescriptionLineBreak !== false) {
+    descriptionLine = `${indent(options)} *${description}
+`
+  }
+  return `${descriptionLine}${indent(options)} * @${options.objectTagName || 'typedef'}${type}${schema.title
     ? ` ${options.capitalizeTitle === false ? schema.title : upperFirst(schema.title)}`
     : ''
   }
 `
 }
 
-function writeParam (type, field, description = '', optional, options) {
-  const fieldTemplate = optional ? `[${field}]` : field
+function writeProperty (type, field, description = '', optional, deflt, options) {
+  let fieldTemplate
+  if (optional) {
+    fieldTemplate = `[${field}${deflt === undefined ? '' : `=${JSON.stringify(deflt)}`}]`
```

Comment:
Why the stringify?

Reply:
If the value in the schema object is a string, for example, it won't have surrounding quotes if it is directly concatenated here. If it is an array, only `toString()` would be called on the array, giving no outer `[` and `]`, etc.

Reply:
And the problem with no surrounding quotes on strings is that a string "true" is not distinguishable from `true` (as with  `false`, `null`, `{}`, `[]`, etc.), made worse if combining the jsdoc output with regular jsdocs (whose default values need not be limited to JSON, e.g., `* @param [win=window]` where "window" could be confusable with `"window"`).